src/app/components/annotations/components/verification-form/verification-parameters.resolver.ts

```diff
@@ -0,0 +1,46 @@
+import { inject, Injectable } from "@angular/core";
+import { ActivatedRouteSnapshot, Resolve } from "@angular/router";
+import { SecurityService } from "@baw-api/security/security.service";
+import { BawSessionService } from "@baw-api/baw-session.service";
+import { lastValueFrom } from "rxjs";
+import { VerificationParameters } from "./verificationParameters";
+
+interface VerificationResolverData {
+  model: VerificationParameters;
+}
+
+@Injectable({ providedIn: "root" })
+class VerificationParametersResolver
+  implements Resolve<VerificationResolverData>
+{
+  private readonly session = inject(BawSessionService);
+  private readonly security = inject(SecurityService);
+
+  public async resolve(
+    route: ActivatedRouteSnapshot,
+  ): Promise<VerificationResolverData> {
+    if (!this.security.firstAuthAwait.value) {
```

Comment:
i know this is here for a reason but i can't remember why.... why do we need auth to deserialize query string parameters?